src/e84_geoai_common/llm/models/claude.py

```diff
@@ -280,4 +268,117 @@ def invoke_model_with_request(self, request: ClaudeInvokeLLMRequest) -> ClaudeRe
         log.info("Token usage: %s", claude_response.usage)
         return claude_response
 
-    # FUTURE implement tool use
+    def response_to_llm_message(
+        self, response: ClaudeResponse, inference_cfg: LLMInferenceConfig
+    ) -> LLMMessage:
+        def _to_llm_content(
+            index: int, c: ClaudeTextContent | ClaudeImageContent | ClaudeToolUseContent
+        ) -> TextContent | Base64ImageContent | LLMToolUseContent:
+            match c:
+                case ClaudeTextContent():
+                    text = c.text
+                    response_prefix = _config_to_response_prefix(inference_cfg)
+                    if index == 0 and response_prefix:
+                        text = response_prefix + text
+                    return TextContent(text=text)
+                case ClaudeImageContent():
+                    return Base64ImageContent(media_type=c.source.media_type, data=c.source.data)
+                case ClaudeToolUseContent():
+                    return LLMToolUseContent(id=c.id, name=c.name, input=c.input)
+
+        return LLMMessage(
+            role="assistant",
+            content=[_to_llm_content(i, c) for i, c in enumerate(response.content)],
+        )
+
+    @staticmethod
+    def llm_message_to_claude_message(msg: LLMMessage) -> "ClaudeMessage":
```

Comment:
It might make more sense to make these module level functions instead of static methods.

Reply:
Changed these to private module-level functions